tests/python/multidevice/test_transformer.py

```diff
@@ -562,492 +543,498 @@ def test_transformer_forward(multidevice_test, benchmark):
-            self.mha_linear0_weight,
-            self.layernorm0_weight,
-            self.layernorm0_mean,
-            self.layernorm0_rstd,
-            self.layernorm0_bias,
-            self.layernorm1_bias,
-        ]:
-            self.sched._set_device_mesh(in_tv, mesh)
+def transformer_backward_multidevice_schedule(fd: FusionDefinition, num_devices: int):
+    mesh = nvfuser.multidevice.DeviceMesh(range(num_devices))
+    inputs = fd.fusion.inputs()
+
+    mlp_linear0_out = inputs[2]
+    out_grad = inputs[3]
+    mlp_linear1_weight = inputs[4]
+    mha_linear1_out = inputs[7]
+    mlp_linear0_weight = inputs[8]
+    layernorm1_weight = inputs[9]
+    layernorm1_mean = inputs[10]
+    inp = inputs[11]
+    layernorm1_rstd = inputs[12]
+    mha_linear1_weight = inputs[13]
+    mha_linear0_out = inputs[14]
+    sdpa_out = inputs[15]
+    sdpa_log_sumexp = inputs[16]
+    mha_linear0_weight = inputs[19]
+    layernorm0_weight = inputs[20]
+    layernorm0_mean = inputs[21]
+    layernorm0_rstd = inputs[22]
+    layernorm0_bias = inputs[23]
+    layernorm1_bias = inputs[24]
 
-        for in_tv in [
-            self.mlp_linear0_out,
-            self.mlp_linear1_weight,
-            self.mlp_linear0_weight,
-            self.mha_linear1_weight,
-            self.mha_linear0_out,
-            self.sdpa_out,
-            self.sdpa_log_sumexp,
-            self.mha_linear0_weight,
-        ]:
-            self.sched.parallelize(in_tv, 0, nvfuser.ParallelType.mesh_x)
+    for in_tv in [
+        mlp_linear0_out,
+        out_grad,
+        mlp_linear1_weight,
+        mha_linear1_out,
+        mlp_linear0_weight,
+        layernorm1_weight,
+        layernorm1_mean,
+        inp,
+        layernorm1_rstd,
+        mha_linear1_weight,
+        mha_linear0_out,
+        sdpa_out,
+        sdpa_log_sumexp,
+        mha_linear0_weight,
+        layernorm0_weight,
+        layernorm0_mean,
+        layernorm0_rstd,
+        layernorm0_bias,
+        layernorm1_bias,
+    ]:
+        in_tv.set_device_mesh(mesh)
+
+    for in_tv in [
+        mlp_linear0_out,
+        mlp_linear1_weight,
+        mlp_linear0_weight,
+        mha_linear1_weight,
+        mha_linear0_out,
+        sdpa_out,
+        sdpa_log_sumexp,
+        mha_linear0_weight,
+    ]:
+        in_tv.axis(0).parallelize(nvfuser.ParallelType.mesh_x)
 
 
 @pytest.mark.skipif(
     is_pre_ampere(),
     reason="Flash Attention is only supported on Ampere and newer devices.",
 )
 @pytest.mark.mpi
-def test_transformer_backward(multidevice_test, benchmark):
-    d = multidevice_test.size
-    rank = multidevice_test.rank
+def test_transformer_backward(multidevice_direct_test, benchmark):
+    d = multidevice_direct_test.size
+    # If only one device exists, then this error occurs:
+    # RuntimeError: unsupported operation: more than one element of the
+    # written-to tensor refers to a single memory location.
+    if d == 1:
+        pytest.skip("This test requires > 1 MPI processes")
```

Comment:
@Priya2698 I remember you were able to run the current test with d==1. Is this a degradation?

Reply:
This ^^^^ is from ToT, so legacy bindings.